controllers/imagejob/imagejob_controller.go

```diff
@@ -253,16 +254,24 @@ func (r *Reconciler) handleNewJob(ctx context.Context, imageJob *eraserv1alpha1.
 	}
 
 	skipped := 0
+	var nodeList []corev1.Node
 
 	log := log.WithValues("job", imageJob.Name)
 
 	env := []corev1.EnvVar{
 		{Name: "NODE_NAME", ValueFrom: &corev1.EnvVarSource{FieldRef: &corev1.ObjectFieldSelector{FieldPath: "spec.nodeName"}}},
 	}
 
-	nodeList, skipped, err := filterOutSkippedNodes(nodes, skipNodesSelectors)
-	if err != nil {
-		return err
+	if *filterOption == "exclude" {
+		nodeList, skipped, err = filterOutSkippedNodes(nodes, filterNodesSelectors)
+		if err != nil {
+			return err
+		}
+	} else if *filterOption == "include" {
+		nodeList, skipped, err = filterIncludedNodes(nodes, filterNodesSelectors)
+		if err != nil {
+			return err
+		}
 	}
```

Comment:
Would be good to have a final `else` branch to return an error in case we have neither `include` nor `exclude` (i.e. a typo or some garbage input). And since we have three branches, might as well use a `switch` block instead.